docs: add recognizing contributor and appointing new maintainers guide (#1964)
asyncapi/community: docs/010-contribution-guidelines/recognizing-contributors-and-appointing-new-maintainers.md

- **Technical Alignment**: Understanding of project structure, workflows, and coding/documentation standards.

### Nomination process

Anyone can nominate a contributor for maintainership, including self-nominations.

1. **Raise a GitHub issue**: In the relevant repo, create a new issue titled `Nomination for Maintainer: @username`.
2. **Summarize contributions**: Include links to PRs, issues, reviews, or community engagement.
3. **Gather feedback**: Allow up to 7 days for community/maintainer feedback on the nomination.
4. **Decision**: Core maintainers or TSC will reach a consensus and respond on the issue.

### Onboarding checklist

When someone is approved as a new maintainer, ensure to do the following:

- [ ] Invite them as a GitHub maintainer with appropriate repo permissions
- [ ] Add them to the [MAINTAINERS.yaml](https://github.com/asyncapi/community/blob/master/MAINTAINERS.yaml) file
- [ ] Share [the Maintainer Onboarding Guide](https://www.asyncapi.com/docs/community/onboarding-guide/maintainer-guide)
- [ ] Schedule a short onboarding call or Slack chat

> - [ ] Encourage joining TSC calls (if applicable)

Reviewer: we don't have these but if they opt to become a TSC member they need to read what is required to be one and the duties

## Recognizing contributors

Contributions come in many forms, not just code. This is how we recognize and celebrate the many ways people help move AsyncAPI forward.

Contributions can include, but are not limited to, any of the following:

- Submitting issues, PRs, or feedback
- Reviewing and testing pull requests
- Writing or editing documentation
- Participating in design discussions
- Creating community content (blogs, talks, tutorials)
- Supporting others in Slack or GitHub issues
- Proposing ideas that influence roadmap or architecture

Basically, if it makes the project better, it counts.

### Recognition process

We use the [All Contributors](https://allcontributors.org/) specification to credit contributors across all AsyncAPI repositories.

To recognize someone:

1. Add a comment on the related issue or PR. For example:

    ```
    @all-contributors please add @username for doc, review
    ```

> 2. Use the appropriate contribution types (like `code`, `doc`, `review`, `ideas`, etc.).

Author: I already added a link to this on line 21 @thulieblack

> 2. Use the appropriate contribution types (like `code`, `doc`, `review`, `ideas`, etc.).

Reviewer 2: I was talking more about having a full list of possible contribution types. Something like that: https://github.com/asyncapi#-contribute-to-asyncapi However, this list lacks explanations on how to get started with each of these contribution types. Also, if we look at the icons that contributors have:

<img width="120" alt="Screenshot 2025-07-06 at 1 10 13 AM" src="https://github.com/user-attachments/assets/088baeca-97e0-4f68-8516-5967a935116e" />

They have even more icons that (despite being self-explanatory to a certain extent) have no proper explanations in docs.

> 2. Use the appropriate contribution types (like `code`, `doc`, `review`, `ideas`, etc.).

Reviewer 2: I think it makes sense to skip my point here and create a follow-up issue to expand several sections of the [Contribution Flow](https://github.com/asyncapi/community/blob/master/docs/010-contribution-guidelines/contribution-flow.md) document, like "Before Making a Contribution" and "Steps to making a good contribution" (step 4)

If you notice a contribution that wasn't acknowledged, speak up in [Slack ](https://asyncapi.com/slack-invite)or [GitHub](https://github.com/asyncapi). We'll gladly fix it.

## Appointing new maintainers

The role of a maintainer is about demonstrating ownership, consistency, and care for the project and its people. In this section, we outline how we identify potential maintainers, the nomination process, and the onboarding steps.

### Selection criteria

Potential maintainers typically demonstrate:

> - **Consistent Contributions**: Quality contributions (code, documentation, reviews, etc.) across multiple PRs/issues.

Reviewer 2: ```suggestion
- **Consistent quality contributions** across multiple PRs/issues.
```

> - **Responsiveness**: Participating in discussions, resolving feedback, and collaborating with others.

Reviewer 2: ```suggestion
- **Responsiveness**: Participating in discussions, resolving feedback, and collaborating.
```

- **Initiative**: Proactively suggesting or leading improvements, not just reacting to tasks.
- **Community Mindset**: Supporting others, offering constructive feedback, and engaging in community spaces (like Slack or streams).
- **Technical Alignment**: Understanding of project structure, workflows, and coding/documentation standards.

### Nomination process

Anyone can nominate a contributor for maintainership, including self-nominations.

> 1. **Raise a GitHub issue**: In the relevant repo, open an issue titled `Nomination for Maintainer: @username`.

Reviewer 2: 'Open' sounds ambiguously as 'open the existing issue named like this'

```suggestion
1. **Raise a GitHub issue**: In the relevant repo, create a new issue titled `Nomination for Maintainer: @username`.
```

2. **Summarize contributions**: Include links to PRs, issues, reviews, or community engagement.

> 3. **Gather feedback**: Allow 5–7 days for community/maintainer feedback on the nomination.

Reviewer 2: ```suggestion
3. **Gather feedback**: Allow up to 7 days for community/maintainer feedback on the nomination.
```

> 4. **Decision**: Core maintainers or TSC will reach a consensus and respond on the issue.

Reviewer 2: Maybe it's a good idea to link to the [TSC definition in CHARTER](https://github.com/asyncapi/community/blob/master/docs/020-governance-and-policies/CHARTER.md#2-technical-steering-committee)

Author: done!

### Onboarding checklist

> When someone is approved as a new maintainer, ensure to do the following:

Reviewer 2: Who is responsible for completing items in this checklist?

Author: Updated

- [ ] Invite them as a GitHub maintainer with appropriate repo permissions
- [ ] Add them to the [MAINTAINERS.yaml](https://github.com/asyncapi/community/blob/master/MAINTAINERS.yaml) file
- [ ] Share [the Maintainer Onboarding Guide](https://www.asyncapi.com/docs/community/onboarding-guide/maintainer-guide)
- [ ] Schedule a short onboarding call or Slack chat
- [ ] Encourage joining TSC calls (if applicable)
- [ ] Introduce them in the relevant Slack channels

> Maintainers are leaders. But leadership isn't about gatekeeping but rather about helping others grow while keeping the project healthy and welcoming. Let's keep celebrating contributions of all shapes and sizes, and support new maintainers as they take on more responsibility.

Reviewer 2: ```suggestion
Maintainers are leaders. But leadership isn't about gatekeeping but rather helping others grow while keeping the project healthy and welcoming. Let's keep celebrating contributions of all shapes and sizes, and support new maintainers as they take on more responsibility.
```

> Have a name in mind? Or would you like to nominate yourself? Head to the relevant repo and open a Maintainer Nomination issue. Let's grow together.

Reviewer 2: ```suggestion
Have a name in mind? Or would you like to nominate yourself? Head to the relevant repo and create a new Maintainer Nomination issue. Let's grow together.
```

# Recognizing Contributors & Appointing New Maintainers

AsyncAPI is built on the passion, ideas, and time of its contributors. From reviewing PRs to improving documentation or running community calls, we recognize that every contribution helps us grow. This document outlines how we recognize contributors and how we appoint and onboard new maintainers across AsyncAPI projects.

## Recognizing contributors

Contributions come in many forms, not just code. This is how we recognize and celebrate the many ways people help move AsyncAPI forward:

### What counts as a contribution?

Contributions can include, but are not limited to, any of the following:

- Submitting issues, PRs, or feedback
- Reviewing and testing pull requests
- Writing or editing documentation
- Participating in design discussions
- Creating community content (blogs, talks, tutorials)
- Supporting others in Slack or GitHub issues
- Proposing ideas that influence roadmap or architecture

> Basically, if it makes the project better, it counts.

Reviewer 3: [suggestion]: I think it would be more beneficial to combine these two parts together to make it more organized

Author: Updated!

### Recognition process

We use the [All Contributors](https://allcontributors.org/) specification to credit contributors across all AsyncAPI repositories.

To recognize someone:

1. Add a comment on the related issue or PR. For example:

    ```
    @all-contributors please add @username for doc, review

>     ```

Author: Hey @CBID2, any reason why you suggest we use inline code blocks instead of multiple line code blocks?

>     ```

Reviewer 3: It's only one line, so I think using three back ticks is best suited for three lines for code @Aahil13.

>     ```

Author: Ok, that's valid.

But seeing that it spans a couple of words, don't you think that, for the purpose of formatting and structure, it'll be best in a multi-code block style?

2. Use the appropriate contribution types (like `code`, `doc`, `review`, `ideas`, etc.).

> If you notice a contribution that wasn't acknowledged, speak up in Slack or GitHub. We'll gladly fix it.

Reviewer 3: ```suggestion
If you notice a contribution that wasn't acknowledged, speak up in [Slack ](https://asyncapi.com/slack-invite)or [GitHub](https://github.com/asyncapi). We'll gladly fix it.
```

### Selection criteria

Potential maintainers typically demonstrate:

- **Consistent Contributions**: Quality contributions (code, documentation, reviews, etc.) across multiple PRs/issues.
- **Responsiveness**: Participating in discussions, resolving feedback, and collaborating with others.
- **Initiative**: Proactively suggesting or leading improvements, not just reacting to tasks.
- **Community Mindset**: Supporting others, offering constructive feedback, and engaging in community spaces (like Slack or streams).
- **Technical Alignment**: Understanding of project structure, workflows, and coding/documentation standards.

### Nomination process

Anyone can nominate a contributor for maintainership, including self-nominations.

1. **Raise a GitHub issue**: In the relevant repo, open an issue titled `Nomination for Maintainer: @username`.
2. **Summarize contributions**: Include links to PRs, issues, reviews, or community engagement.
3. **Gather feedback**: Allow 5–7 days for community/maintainer feedback on the nomination.
4. **Decision**: Core maintainers or TSC will reach a consensus and respond on the issue.

### Onboarding checklist

> When someone is approved as a new maintainer:

Reviewer 3: ```suggestion
When someone is approved as a new maintainer, ensure to do the following:
```

- [ ] Invite them as a GitHub maintainer with appropriate repo permissions
- [ ] Add them to the [MAINTAINERS.yaml](https://github.com/asyncapi/community/blob/master/MAINTAINERS.yaml) file

> - [ ] Share the Maintainer Onboarding Guide

Reviewer 3: ```suggestion
- [ ] Share [the Maintainer Onboarding Guide](https://www.asyncapi.com/docs/community/onboarding-guide/maintainer-guide)
```
